apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx

```diff
@@ -101,10 +118,11 @@ const FinalStep: React.FunctionComponent< LaunchStepProps > = ( { onNextStep, on
 
 	const planSummary = (
 		<div className="nux-launch__summary-item">
-			{ plan && ! plan?.isFree ? (
+			{ plan && planProduct && ! isPlanFree ? (
 				<>
 					<p className="nux-launch__summary-item__plan-name">WordPress.com { plan.title }</p>
-					{ __( 'Plan subscription', 'full-site-editing' ) }: { planPrices[ plan.storeSlug ] }{ ' ' }
+					{ __( 'Plan subscription', 'full-site-editing' ) }:{ ' ' }
+					{ planPrices[ planProduct.storeSlug ] }{ ' ' }
```

Comment:
Since you now have the planProduct, you can do 
```suggestion
					{ planProduct.price }{ ' ' }
```